test/asn1_time_test.c

```diff
@@ -79,7 +79,7 @@ static struct testdata tbl_testdata_pos[] = {
 
 /* ASSUMES SIGNED TIME_T */
 static struct testdata tbl_testdata_neg[] = {
-    { "19011213204552Z",   V_ASN1_GENERALIZEDTIME, V_ASN1_GENERALIZEDTIME, 1, -2147483648, -1, 0, },
+    { "19011213204552Z",   V_ASN1_GENERALIZEDTIME, V_ASN1_GENERALIZEDTIME, 1, (time_t)-1*2147483648, -1, 0, },
```

Comment:
Is this done like this to avoid overflow?  Space around the multiplier operation.  Perhaps a comment explaining why it's written like this?

Reply:
Yeah, about that... Problem there is that compiler complains with "unary minus operator applied to unsigned type, result still unsigned". But target type is actually signed, and from standard's point of view unsigned to signed is implementation-specific, which in turn something we would like to avoid. Hence the trouble. One can probably argue that it's kind of a compiler bug, because -123 can be viewed not as unsigned with unary minus, but as negative number to start with. But now after extra consideration I've found alternative, less controversial way to solve it.

[As curious fact, INT_MIN is customarily defined as -2147483647-1, so that compiler in question doesn't complain, because 2147483647 doesn't have most significant bit set.]